src/connection.h

```diff
@@ -68,6 +82,7 @@ typedef void (*ConnectionCallbackFunc)(struct connection *conn);
 
 typedef struct ConnectionType {
     /* connection type */
+    int (*get_type_id)(struct connection *conn);
     const char *(*get_type)(struct connection *conn);
```

Comment:
I don't think `get_type` is needed anymore. we only need `get_type` today for logging so a simple helper function that maps a given type id to its string representation would suffice.

Reply:
thanks @artikell. separate PR is good.

Reply:
filed #2187 to track this cleanup work